polkadot/runtime/parachains/src/hrmp/benchmarking.rs

```diff
@@ -138,77 +141,99 @@ static_assertions::const_assert!(MAX_UNIQUE_CHANNELS < PREFIX_0);
+		let deposit: BalanceOf<T> =
+			Configuration::<T>::config().hrmp_sender_deposit.unique_saturated_into();
 		register_parachain_with_balance::<T>(sender_id, deposit + ed);
 		register_parachain_with_balance::<T>(recipient_id, deposit + ed);
 
 		let capacity = Configuration::<T>::config().hrmp_channel_max_capacity;
 		let message_size = Configuration::<T>::config().hrmp_channel_max_message_size;
-	}: _(sender_origin, recipient_id, capacity, message_size)
-	verify {
+
+		#[extrinsic_call]
+		_(sender_origin, recipient_id, capacity, message_size);
+
 		assert_last_event::<T>(
-			Event::<T>::OpenChannelRequested(sender_id, recipient_id, capacity, message_size).into()
+			Event::<T>::OpenChannelRequested(sender_id, recipient_id, capacity, message_size)
+				.into(),
 		);
 	}
 
-	hrmp_accept_open_channel {
+	#[benchmark]
+	fn hrmp_accept_open_channel() {
 		let [(sender, _), (recipient, recipient_origin)] =
 			establish_para_connection::<T>(1, 2, ParachainSetupStep::Requested);
-	}: _(recipient_origin, sender)
-	verify {
+
+		#[extrinsic_call]
+		_(recipient_origin, sender);
+
 		assert_last_event::<T>(Event::<T>::OpenChannelAccepted(sender, recipient).into());
 	}
 
-	hrmp_close_channel {
+	#[benchmark]
+	fn hrmp_close_channel() {
 		let [(sender, sender_origin), (recipient, _)] =
 			establish_para_connection::<T>(1, 2, ParachainSetupStep::Established);
 		let channel_id = HrmpChannelId { sender, recipient };
-	}: _(sender_origin, channel_id.clone())
-	verify {
+
+		#[extrinsic_call]
+		_(sender_origin, channel_id.clone());
+
 		assert_last_event::<T>(Event::<T>::ChannelClosed(sender, channel_id).into());
 	}
 
 	// NOTE: a single parachain should have the maximum number of allowed ingress and egress
 	// channels.
-	force_clean_hrmp {
+	#[benchmark]
+	fn force_clean_hrmp(
 		// ingress channels to a single leaving parachain that need to be closed.
-		let i in 0 .. (HRMP_MAX_INBOUND_CHANNELS_BOUND - 1);
+		i: Linear<0, { HRMP_MAX_INBOUND_CHANNELS_BOUND - 1 }>,
 		// egress channels to a single leaving parachain that need to be closed.
-		let e in 0 .. (HRMP_MAX_OUTBOUND_CHANNELS_BOUND - 1);
-
+		e: Linear<0, { HRMP_MAX_OUTBOUND_CHANNELS_BOUND - 1 }>,
+	) {
 		// first, update the configs to support this many open channels...
-		assert_ok!(
-			Configuration::<T>::set_hrmp_max_parachain_outbound_channels(frame_system::RawOrigin::Root.into(), e + 1)
-		);
-		assert_ok!(
-			Configuration::<T>::set_hrmp_max_parachain_inbound_channels(frame_system::RawOrigin::Root.into(), i + 1)
-		);
+		assert_ok!(Configuration::<T>::set_hrmp_max_parachain_outbound_channels(
+			frame_system::RawOrigin::Root.into(),
+			e + 1
+		));
+		assert_ok!(Configuration::<T>::set_hrmp_max_parachain_inbound_channels(
+			frame_system::RawOrigin::Root.into(),
+			i + 1
+		));
+		assert_ok!(Configuration::<T>::set_max_downward_message_size(
+			frame_system::RawOrigin::Root.into(),
+			1024
+		));
```

Comment:
I'm a little surprised this works since the function [just schedules](https://github.com/paritytech/polkadot-sdk/blob/1e2a2f0/polkadot/runtime/parachains/src/configuration.rs#L879) the change. But I also ran this benchmark locally and it does work.

Reply:
In the existing code, after the changes were scheduled, `Configuration::<T>::initializer_on_new_session` was used, which [activates it](https://github.com/paritytech/polkadot-sdk/blob/1e2a2f0c69f3d8179a45363ddad18d26b5531183/polkadot/runtime/parachains/src/configuration.rs#L1255).

Reply:
Oh yeah, my bad.